src/tests/tests/background.js

```diff
@@ -119,4 +124,22 @@
     }
   });
 
+  QUnit.test("DNT checking obeys user setting", (assert) => {
+    const NUM_TESTS = 5;
+
+    let done = assert.async(NUM_TESTS);
+
+    assert.expect(NUM_TESTS);
+
+    badger.isCheckingDNTPolicyEnabled = () => false;
```

Comment:
I've been thinking about this one. I think this needs to be more generalized for the state of the local storage in all unit tests. The thing that happens in the storage.js test should happen in all tests. All unit tests should save the local storage, replace with a standard one, and per test change it depending on the thing they are looking at. I think this wasn't done before because I think all the unit tests just avoided messing with anything that changed with settings.